src/api/auth/LighthouseAuthApi.ts

```diff
@@ -76,19 +120,47 @@ export class LighthouseAuthApi implements AuthApi {
       method: 'POST',
       credentials: 'include',
     });
-
     return true;
   }
 
   async getPublicUsers(): Promise<User[]> {
-    throw new Error('Method not implemented.');
+    // TODO: we currently don't have the concept of public users in the new API (heimdall)
+    return this.getAllUsers();
   }
 
   async getAllUsers(): Promise<User[]> {
-    throw new Error('Method not implemented.');
+    const apiUsersResponse = await fetch(`${this.url}/users`, {
+      credentials: 'include',
+    });
+    const apiUsers: ApiUser[] = await apiUsersResponse.json();
+    const users: User[] = apiUsers.map(apiUser => {
+      const user: User = {
+        username: apiUser.username,
+        role: undefined, // TODO: change role to roles
+        course: apiUser.registration_key?.key,
+        createdAt: new Date(apiUser.created_at),
+        lastSeen: new Date(apiUser.last_login),
+      };
```

Comment:
I suggest removing the mapping entirely for simplicity and just returning the "ApiUser" as the User, since we need the fields in the rest of the application at a later point. And the legacy integration could be removed or altered to fit the new API, since luna is supposed to be the new frontend for the new backend and not the old one :smile:
But it's your project, so you decide how to build it :wink: